roles/postgres/files/rotate.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/python
+
+"""Rotate backup files in a given directory.
+
+Implements a primitive FIFO backup rotation strategy by keeping N most recent backup files. The
+order is defined by sorting the file names lexicographically: the files that appear later in
+the sorted list are considered to be newer.
+"""
+
+import argparse
+import dataclasses
+import pathlib
+
+
+@dataclasses.dataclass(order=True)
+class Backup:
+    path: pathlib.Path
+    size: int
+
+
+@dataclasses.dataclass
+class Args:
+    keep: int
+    dir: pathlib.Path
+    pattern: str
+    nodry_run: bool
+
+
+def non_negative_int(str_value: str) -> int:
+    value = int(str_value)
+    if value < 0:
+        raise argparse.ArgumentTypeError(f"Value must be non-negative: {value} < 0")
+
+    return value
+
+
+def parse_args() -> Args:
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument(
+        "-n",
+        "--keep",
+        required=True,
+        type=non_negative_int,
+        help="Keep this many most recent backup files",
+    )
+    parser.add_argument(
+        "-d", "--dir", type=pathlib.Path, help="Path to the directory with backup files"
+    )
+    parser.add_argument(
+        "-p",
+        "--pattern",
+        type=str,
+        default="*",
+        help="Only consider files that match this glob pattern",
+    )
+    parser.add_argument(
+        "--nodry_run",
```

Comment:
nit: the parameter can contain dashes, i.e. `--no-dry-run`, and it will generate the attribute `no_dry_run` in the namespace object automatically. I think dashes in parameter names are more natural when it comes to command line interfaces.

Reply:
Heh, you can tell that I haven't used argparse in quite some time now! :)